src/switch_core_media.c

```diff
@@ -11302,6 +11329,34 @@ SWITCH_DECLARE(void)switch_core_media_set_local_sdp(switch_core_session_t *sessi
 	if (smh->sdp_mutex) switch_mutex_unlock(smh->sdp_mutex);
 }
 
+static SWITCH_DECLARE(switch_status_t) add_rtp_header_extensions(switch_core_session_t *session, sdp_media_e media, char *buf, uint32_t buflen)
+{
+	const char *val;
+	int extmap_id = 1;
+
+	if ((val = switch_channel_get_variable(session->channel, "audio_level_events")) && switch_true(val)) {
```

Comment:
I would suggest to put this outside of this function. In switch_core_media_gen_local_sdp, you will have to check these channel variable `audio_level_events` before calling the add_rtp_header extension.

Reply:
it is because this is one of the RTP extensions, if we implement others, we need to create other feature flippers and has an if for the other extension.

Reply:
maybe we can create other function for this, and put something like
```
if (...)
add_rtp_header_extension_audio_level_event()
```

Reply:
This is merely a suggestion. I would prefer to change the `audio_level_events` to `rtp_ext_audio_level_events` so we know that this variable is related to media.

Reply:
Agreed. Also prefix it with _telnyx__.

Reply:
Regarding the method to group adding of RTP extensions, I think it makes sense to have `add_rtp_header_extensions` method in `switch_core_media_gen_local_sdp`.